synapseclient/models/protocols/json_schema_protocol.py

```diff
@@ -0,0 +1,135 @@
+        """
+
+    def validate_schema(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> Union[JSONSchemaValidation, InvalidJSONSchemaValidation]:
+        """
+        Validate the entity against the bound JSON schema.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            Union[JSONSchemaValidation, InvalidJSONSchemaValidation]: The validation results.
+        """
+        return InvalidJSONSchemaValidation() or JSONSchemaValidation()
+
+    def get_schema_derived_keys(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> JSONSchemaDerivedKeys:
+        """
+        Retrieve derived JSON schema keys for the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaDerivedKeys: An object containing the derived keys for the entity.
+        """
+        return JSONSchemaDerivedKeys()
+
+
+class ContainerEntityJSONSchemaProtocol(BaseJSONSchemaProtocol):
+    """
+    Mixin class to provide JSON schema functionality for container entities.
+    This class extends BaseJSONSchemaProtocol and is intended for use with Synapse container entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the container entity.
+    """
+
+    async def get_schema_validation_statistics_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> JSONSchemaValidationStatistics:
+        """
+        Get validation statistics for a container entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaValidationStatistics: The validation statistics.
+        """
+        return JSONSchemaValidationStatistics()
+
+    async def get_invalid_validation_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> AsyncGenerator[InvalidJSONSchemaValidation, None]:
+        """
+        Get invalid JSON schema validation results for a container entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Yields:
+            InvalidJSONSchemaValidation: An object containing the validation response, all validation messages,
+                                         and the validation exception details.
+        """
+        return InvalidJSONSchemaValidation(), None
```

Comment:
@BryanFauble this one involves the async generator. Could you double check if this is correct?